### DISCUSSION TITLE
Hardware monitoring API discussion

### DISCUSSION THREAD

**Discussion by rocksdanister** [rocksdanister/lively#233] [Ideas]
Lets discuss about the upcoming Hardware monitor api https://github.com/rocksdanister/lively/issues/152

Currently these are the supported data:
![Screenshot 2020-12-13 085638](https://user-images.githubusercontent.com/17554161/102002405-3a3a1700-3d22-11eb-9844-ae5555750e61.png)
(More data will be added as this discussion continues.)

The data will be sent in JSON format, to read it in javascript:
```js
function livelySystemInformation(data)
{
    var obj = JSON.parse(data);
    console.log(obj.NameCpu + " " + obj.CurrentCpu);
    console.log(obj.NameGpu + " " + obj.CurrentGpu3D);
    console.log("RAM(Total): " + obj.TotalRam);
    console.log("RAM(Free): " + obj.CurrentRamAvail);
    console.log("NETDOWN(MB/s): " + obj.CurrentNetDown/(1024*1024));
    console.log("NETUP: " + obj.CurrentNetUp);
}
```
Feedback and suggestions are welcome!

**Comment by rocksdanister** [comment]
Added network cardname to the data.
Updated doc:
https://github.com/rocksdanister/lively/wiki/Web-Guide-V-:-System-Data

**Comment by rocksdanister** [comment]
Beta build is up:
https://github.com/rocksdanister/lively-beta/releases/tag/v1.1.6.1

**Comment by rocksdanister** [comment]
Working on a simple demo wallpaper

https://user-images.githubusercontent.com/17554161/102693919-e492cc80-4243-11eb-8941-42d076cb3080.mp4

**Comment by Darkaxt** [comment]
Would it be possible to return the actual in use ram, rather than the committed one? I have combined this wallpaper, with my custom laptop model and the wallpaper "The Spirit" and this is the last remaining challenge.

![image](https://github.com/user-attachments/assets/7b5cab4e-711a-497b-af54-42a66d1cf486)

**Reply by rocksdanister** [reply]
that looks beautiful.

As for the memory, didn't think about that. The API is in need of an update to support multiple gpu and network cards, will look into it.
